mux.go

```diff
@@ -59,6 +59,12 @@ func (r *Router) Match(req *http.Request, match *RouteMatch) bool {
 			return true
 		}
 	}
+
+	// Closest match for a router (includes sub-routers)
+	if r.NotFoundHandler != nil {
+		match.Handler = r.NotFoundHandler
+		return true
```

Comment:
Does it really make sense to report this as a match?
http://play.golang.org/p/SIL-WWftyQ
Seems confusing when request didn't match any real route but `router.Match` reports `true`.

FWIW I got a panic in my stats collection code which assumed that if `router.Match(...) == true`, returned `RouteMatch` would have a non-nil `Route`.